### DISCUSSION TITLE
How to use topLayer with @floating-ui/vue

### DISCUSSION THREAD

**Discussion by antoniogiroz** [floating-ui/floating-ui#3176] [Q&A]
I'm using @floating-ui/vue (1.1.5) to create a tooltip component.

I'm using it with a button within a native dialog element, and the tooltip is shown behind the dialog. I've tried to use the platform object as I've seen in this PR https://github.com/floating-ui/floating-ui/pull/2747

However the vue composable does not have the `platform` attribute.

Do you know how I could solve this?

Thanks!

**Comment by atomiks** [comment]
The tooltip either also needs to be a top-layer element, or be mounted inside the dialog element itself so it's also top-layer.

**Reply by antoniogiroz** [reply]
I've prepared a CodeSandbox with a little example https://codesandbox.io/p/devbox/3s8t5z.

The `<Teleport to="body">` positions the tooltip behind the dialog. Without it, the tooltip appears directly on top of the dialog, and the boundaries are restricted to those of the dialog, causing a scrollbar to appear 😓

**Reply by FezVrasta** [reply]
> I've prepared a CodeSandbox with a little example https://codesandbox.io/p/devbox/3s8t5z.
> 
> 
> 
> The `<Teleport to="body">` positions the tooltip behind the dialog. Without it, the tooltip appears directly on top of the dialog, and the boundaries are restricted to those of the dialog, causing a scrollbar to appear 😓

Have you tried making the tooltip a top layer element as well?

**Reply by antoniogiroz** [reply]
Hmm, how can I do that? 🤔 Using a popover? Do you have an example?

Anyway, it is supposed that this library solves this issue. isn't?

**Reply by andrewpitts15** [reply]
Did you ever solve this issue? I'm having the same problem.

**Reply by atomiks** [reply]
This is just related to CSS top-layer elements. You need to make sure the tooltip is also a top-layer element (the new `popover` attribute is one way) in order to appear above the dialog. Otherwise, it must be inside the dialog itself

It's not related to this library (which solves anchoring, not layering)

**Reply by andrewpitts15** [reply]
So does the PR https://github.com/floating-ui/floating-ui/pull/2747 only solve the issue for vanilla javascript implementations of the library and not the Vue and React implementations? 

Can you elaborate on your suggestion of adding the popover attribute? When I add it to the tooltip it doesn't seem to make a difference.

**Reply by atomiks** [reply]
That PR solves positioning issues not layering issues. I just assumed native top-layer stuff worked correctly (in general on the web, not in this lib), but it turns out the API itself has numerous issues regarding nested top-layer elements - [take a look here](https://benfrain.com/failing-with-multiple-dialog-elements-understanding-the-top-layer-and-popovers/)

Top-layer is a bad idea (right now): https://github.com/mui/base-ui/issues/485#issuecomment-2241703553
